features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/Listing/Edit.jsx

```diff
@@ -0,0 +1,305 @@
+import TextField from '@material-ui/core/TextField';
+import { makeStyles } from '@material-ui/core/styles';
+import ServiceCatalog from 'AppData/ServiceCatalog';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemText from '@material-ui/core/ListItemText';
+
+const useStyles = makeStyles((theme) => ({
+    mandatoryStar: {
+        color: theme.palette.error.main,
+        marginLeft: theme.spacing(0.1),
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state State.
+ * @returns {Promise} Promised state.
+ */
+function reducer(state, { field, value }) {
+    switch (field) {
+        case 'name':
+            return { ...state, [field]: value, displayName: value };
+        case 'serviceUrl':
+            return { ...state, [field]: value };
+        case 'definitionType':
+            return { ...state, [field]: value };
+        default:
+            return state;
+    }
+}
+
+/**
+* Service Catalog service edit
+* @param {any} props Props for edit function.
+* @returns {any} Returns the rendered UI for service edit.
+*/
+function Edit(props) {
+    const intl = useIntl();
+    const classes = useStyles();
+    const { getData, dataRow } = props;
+    const [open, setOpen] = useState(false);
+    const [schemaTypeList, setSchemaTypeList] = useState([]);
+    const [serviceTypeList, setServiceTypeList] = useState([]);
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+    const [initialState, setInitialState] = useState(dataRow);
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const {
+        id,
+        name,
+        serviceUrl,
+        definitionType,
+    } = state;
+
+    useEffect(() => {
+        setInitialState(dataRow);
+        const settingPromise = ServiceCatalog.getSettings();
+        // Take from settings when backend API is implemented
+        settingPromise.then(() => {
+            setSchemaTypeList(['OAS2', 'OAS3', 'WSDL1', 'WSDL2', 'GRAPHQL_SDL', 'ASYNC_API']);
+            setServiceTypeList(['OAS2', 'OAS3', 'WSDL1', 'WSDL2', 'GRAPHQL_SDL', 'ASYNC_API']);
+        }).catch(() => {
+            Alert.error(intl.formatMessage({
+                defaultMessage: 'Error while retrieving service and schema types',
+                id: 'ServiceCatalog.Listing.Edit.error.retrieve.service.schema.types',
+            }));
+        });
+    }, []);
+
+    const handleChange = (e) => {
+        dispatch({ field: e.target.name, value: e.target.value });
+    };
+
+    /**
+     * Function for updating a given service entry
+     */
+    function doneEditing() {
+        const updateServicePromise = ServiceCatalog.updateService(id, state);
+        updateServicePromise.then(() => {
+            Alert.info(intl.formatMessage({
+                id: 'ServiceCatalog.Listing.Delete.service.updated.successfully',
+                defaultMessage: 'Service updated successfully!',
+            }));
+            // Reload the services list
+            getData();
```

Comment:
As mention above I think it's better to invoke a `onEdit` function here and pass this promise `updateServicePromise` object to it, So it can handle what to do in case of a failure or success save